doc/nrf/releases/release-notes-1.5.0.rst

```diff
@@ -0,0 +1,1132 @@
+
+* Build system:
+
+  * Ensured that shields can be placed in other BOARD_ROOT folders.
+  * Added basic support for Clang 10 with x86.
+  * Fixed a bug that prevented compiling the :ref:`bootloader` with :option:`CONFIG_SB_SIGNING_PUBLIC_KEY`
+
+* System:
+
+  * Added an API that provides a printf family of functions (for example, :c:func:`cbprintf`) with a callback on character output, to perform in-place streaming of the formatted string.
+  * Updated minimal libc to print stderr just like stdout.
+  * Added an ``abort()`` function to minimal libc.
+  * Updated the ring buffer to allow using the full buffer capacity instead of forcing an empty slot.
+  * Added a :c:macro:`CLAMP` macro.
+  * Added a feature for post-mortem analysis to the tracing library.
+
+* Samples:
+
+  * Added :ref:`zephyr:nrf-ieee802154-rpmsg-sample`.
+  * Added :ref:`zephyr:cloud-tagoio-http-post-sample`.
+  * Added :ref:`zephyr:civetweb-websocket-server-sample`.
+  * :ref:`zephyr:led_ws2812_sample`: Updated to force SPIM on nRF52 DK.
+  * :ref:`zephyr:cfb_custom_fonts`: Added support for ssd1306fb.
+  * :ref:`zephyr:gsm-modem-sample`: Added suspend/resume shell commands.
+  * :ref:`zephyr:updatehub_fota_sample`: Added support for |BLE| IPSP, 802.15.4, modem, and Wi-Fi.
+
+* Logging:
+
+  * Added STP transport and raw data output support for systrace.
+
+* Modules:
+
+  * Introduced a :option:`CONFIG_MBEDTLS_MEMORY_DEBUG` option for mbedtls.
+  * Updated LVGL to v7.6.1.
+  * Updated libmetal and openamp to v2020.10.
+  * Updated nrfx in hal-nordic to version 2.4.0.
+  * Updated the Trusted Firmware-M (TF-M) module to v1.2.0.
+  * Moved the nrfx glue code from the hal_nordic module repository to the main Zephyr repository.
+  * Updated the Trusted Firmware-M (TF-M) module to include support for the nRF5340 and nRF9160 platforms.
+
+* Other:
+
+  * Renamed the ``sanitycheck`` script to ``twister``.
+  * Added initial LoRaWAN support.
+  * Updated ``west flash`` support for ``nrfjprog`` to fail if a HEX file has UICR data and ``--erase`` was not specified.
+  * Added an API to correlate system time with external time sources (see :ref:`zephyr:timeutil_api`).
+
+* Power management:
+
+  * Overhauled the naming and did some general cleanup.
+  * Added a notifier API to register an object to receive notifications when the system changes power state.
+
+* Shell:
+
+  * Updated documentation.
+  * Optimized the tab feature and the select command.
+  * Enhanced and improved the help command.
+
+* Toolchain:
+
+  * Added initial support for LLVM/Clang (version 10, on the x86 architecture).
+  * Added the environment variable ``LLVM_TOOLCHAIN_PATH`` for locating the LLVM toolchain.
+
+* USB:
+
+  * Fixed the handling of zero-length packet (ZLP) in the nRF USB Device Controller Driver.
+  * Changed the USB DFU wait delay to be configurable with Kconfig (:option:`CONFIG_USB_DFU_WAIT_DELAY_MS`).
+
+Additions specific to |NCS|
++++++++++++++++++++++++++++
```

Comment:
@mbolivar-nordic Is this section (copied from v1.4.0) still correct and up-to-date?

Reply:
>  @mbolivar-nordic Is this section (copied from v1.4.0) still correct and up-to-date?

@ru-fu instead of answering your question directly I will enlist the help of the appropriate people, as I can't say for sure in all cases.

Here is a list of NCS-only zephyr patches, excluding `fromlist` and `fromtree` since those are at least spiritually not NCS specific.

I broke them down by area and added `@`-mentions to the people I think should confirm that these notes are up to date with respect to these commits.

Build system (@tejlmand):

- 46e99f3de4 [nrf temphack] cmake: exclude print-ot-config for target ALL
- 429476d105 [nrf temphack] setting `ZEPHYR_<MODULE_NAME>_KCONFIG` for NCS modules
- 0332aacb8e [nrf temphack] Revert "modules: tfm: cleanup redundant Kconfig...
- 09b7569da1 [nrf temphack] runners: nrfjprog: hackaround NCSDK-7327
- 399915c953 [nrf noup] tree-wide: support NCS Partition Manager (PM) definitions
- 10202d4b71 [nrf temphack] boards: arm: nrf5340: disable TF-M on non-secure builds

Devicetree (@mbolivar-nordic; I will suggest a change to cover these in a separate comment):

- 59c5ac83cc [nrf noup] dts: add binding file for cc132 cryptocell sw
- 0c02197497 [nrf noup] dts: Adding cryptocell_sw as chosen entropy
- a8e33475f9 [nrf noup] dts: prefer CC310 for zephyr,entropy
- 6f5b6f4107 [nrf noup] dts: add compatibles for CC310 NS access

Drivers (@ioannisg):

- 7317a1ccc3 [nrf toup] drivers: clock_control: nrf: Add assert on unexpected event
- a2dfd93da7 [nrf toup] clock_control: nrf: modify Kconfig accuracy
- ae0e993299 [nrf toup] clock_control: nrf: Change CLOCK_CONTROL_NRF_FORCE_ALT

Networking (@rlubos):

- d09f04c496 [nrf noup] drivers: ieee802154: SPM load EUI64 from UICR
- 17948b7158 [nrf noup] drivers: ieee802154: Fix build error on nRF5340
- 0ab7df74b2 [nrf noup] drivers: ieee802154: Use nRF Secure Services to read EUI64
- 3ebae5387c [nrf noup] net: l2: Kconfig: Add option for Zigbee L2 layer
- 23d4adb64f [nrf noup] net: lib: mqtt: Provide option to enable TLS session caching
- 38a90a7768 [nrf noup] include: net: add NCS extensions

Testing (@thst-nordic):

- e18c99de35 [nrf noup] ci: Compliance workflow downstream tweaks
- 654ac08743 [nrf noup] ci: Add Jenkins-based CI for zephyr

Documentation (@ru-fu):

- 7bedae6972 [nrf noup] doc: Share Kconfig documentation between repos

Bluetooth (@cvinayak or @joerchan):

- 2ddfb47280 [nrf noup] samples: smp_svr: Increase stack size for overlay-bt.conf
- 59a35645b6 [nrf noup] bluetooth: controller: default to SPLIT architecture

Reply:
I have asked the relevant team members to take a look at this. Some of these `[nrf noup]` are false positives (i.e. they are indeed upstream).

Reply:
Regarding the Testing and Documentation sections, nothing should be added here, they do not need to be documented in any way, since they do not affect the codebase itself.

Reply:
For Bluetooth I would say there is nothing that should be added of those two. One just increases stack size to fit SD controller. Default to SPLIT changes nothing.

Reply:
For drivers, I don't see anything in those 3 commits above that should be mentioned in release notes.

Reply:
OK, I am closing this then. I will approve the PR, thank you all.